[create] panic: runtime error: invalid memory address or nil pointer dereference

```bash
root@xyz:~$ dokku elasticsearch:destroy snowstorm-es
 !     WARNING: Potentially Destructive Action
 !     This command will destroy snowstorm-es Elasticsearch service.
 !     To proceed, type "snowstorm-es"

> snowstorm-es
=====> Deleting snowstorm-es
=====> Stopping container
       Container stopped
       Removing container
       Removing data
=====> Elasticsearch container deleted: snowstorm-es
root@xyz~$ dokku elasticsearch:create snowstorm-es
=====> Extracting config files
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x562f7fb377f8]

goroutine 1 [running]:
github.com/docker/cli/cli/command.ValidateOutputPath(0x7ffe2f194085, 0x1, 0xc0002b4180, 0x20)
        /go/src/github.com/docker/cli/cli/command/utils.go:142 +0x118
github.com/docker/cli/cli/command/container.copyFromContainer(0x562f815d18c0, 0xc00005a128, 0x562f81613cc0, 0xc0003460f0, 0x0, 0x7ffe2f194064, 0x20, 0x7ffe2f194085, 0x1, 0x7ffe2f194023, ...)
        /go/src/github.com/docker/cli/cli/command/container/cp.go:131 +0xd2
github.com/docker/cli/cli/command/container.runCopy(0x562f81613cc0, 0xc0003460f0, 0x7ffe2f194023, 0x61, 0x7ffe2f194085, 0x1, 0x0, 0x0, 0xc000035c58)
        /go/src/github.com/docker/cli/cli/command/container/cp.go:102 +0x3b4
github.com/docker/cli/cli/command/container.NewCopyCommand.func1(0xc0001d4f00, 0xc0002b4140, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/docker/cli/cli/command/container/cp.go:67 +0xe3
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc0001d4f00, 0xc0001c5f90, 0x2, 0x2, 0xc0001d4f00, 0xc0001c5f90)
        /go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:762 +0x467
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0002c7180, 0xc0001c5f80, 0x3, 0x3)
        /go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:852 +0x2ee
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:800
main.runDocker(0xc0003460f0, 0x562f8157c440, 0xc000010020)
        /go/src/github.com/docker/cli/cmd/docker/docker.go:281 +0x1d5
main.main()
        /go/src/github.com/docker/cli/cmd/docker/docker.go:292 +0xf3
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
```

```bash 
root@xyz$ export ELASTICSEARCH_IMAGE="elasticsearch"
root@xyz$ export ELASTICSEARCH_IMAGE_VERSION="6.8.2"
root@xyz$ dokku elasticsearch:destroy snowstorm-es
 !     WARNING: Potentially Destructive Action
 !     This command will destroy snowstorm-es Elasticsearch service.
 !     To proceed, type "snowstorm-es"

> snowstorm-es
=====> Deleting snowstorm-es
 !     Service is already stopped
       Removing data
=====> Elasticsearch container deleted: snowstorm-es
root@xyz$ dokku elasticsearch:create snowstorm-es
=====> Extracting config files
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x558d1366e7f8]

goroutine 1 [running]:
github.com/docker/cli/cli/command.ValidateOutputPath(0x7ffe9f409085, 0x1, 0xc0005e9460, 0x20)
        /go/src/github.com/docker/cli/cli/command/utils.go:142 +0x118
github.com/docker/cli/cli/command/container.copyFromContainer(0x558d151088c0, 0xc0000ca020, 0x558d1514acc0, 0xc0002140f0, 0x0, 0x7ffe9f409064, 0x20, 0x7ffe9f409085, 0x1, 0x7ffe9f409023, ...)
        /go/src/github.com/docker/cli/cli/command/container/cp.go:131 +0xd2
github.com/docker/cli/cli/command/container.runCopy(0x558d1514acc0, 0xc0002140f0, 0x7ffe9f409023, 0x61, 0x7ffe9f409085, 0x1, 0x0, 0x0, 0xc000271c58)
        /go/src/github.com/docker/cli/cli/command/container/cp.go:102 +0x3b4
github.com/docker/cli/cli/command/container.NewCopyCommand.func1(0xc0005ce280, 0xc0005e9420, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/docker/cli/cli/command/container/cp.go:67 +0xe3
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc0005ce280, 0xc0005ded60, 0x2, 0x2, 0xc0005ce280, 0xc0005ded60)
        /go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:762 +0x467
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003c6f00, 0xc0005ded50, 0x3, 0x3)
        /go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:852 +0x2ee
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:800
main.runDocker(0xc0002140f0, 0x558d150b3440, 0xc0000cc010)
        /go/src/github.com/docker/cli/cmd/docker/docker.go:281 +0x1d5
main.main()
        /go/src/github.com/docker/cli/cmd/docker/docker.go:292 +0xf3
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
```

The issue is present with Docker `19.03.5`.

```
root@xyz:~# dokku elasticsearch:start statica
=====> Starting container
=====> Extracting config files
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x55750ed54cf8]

goroutine 1 [running]:
github.com/docker/cli/cli/command.ValidateOutputPath(0x7ffeaa7b7716, 0x1, 0xc000382d80, 0x20)
	/go/src/github.com/docker/cli/cli/command/utils.go:142 +0x118
github.com/docker/cli/cli/command/container.copyFromContainer(0x5575105f2d00, 0xc0000520d8, 0x5575106351a0, 0xc0000d20f0, 0x0, 0x7ffeaa7b76f5, 0x20, 0x7ffeaa7b7716, 0x1, 0x7ffeaa7b76b4, ...)
	/go/src/github.com/docker/cli/cli/command/container/cp.go:131 +0xd2
github.com/docker/cli/cli/command/container.runCopy(0x5575106351a0, 0xc0000d20f0, 0x7ffeaa7b76b4, 0x61, 0x7ffeaa7b7716, 0x1, 0x0, 0x0, 0xc00025dc58)
	/go/src/github.com/docker/cli/cli/command/container/cp.go:102 +0x3b4
github.com/docker/cli/cli/command/container.NewCopyCommand.func1(0xc000230a00, 0xc000382d40, 0x2, 0x2, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/container/cp.go:67 +0xe3
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc000230a00, 0xc000246340, 0x2, 0x2, 0xc000230a00, 0xc000246340)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:762 +0x467
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0000ce500, 0xc000246330, 0x3, 0x3)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:852 +0x2ee
github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:800
main.runDocker(0xc0000d20f0, 0x55751059d840, 0xc000010028)
	/go/src/github.com/docker/cli/cmd/docker/docker.go:281 +0x1d5
main.main()
	/go/src/github.com/docker/cli/cmd/docker/docker.go:292 +0xf3
tar: This does not look like a tar archive
tar: Exiting with failure status due to previous errors
```

Dokku report:

```
-----> uname: Linux xyz 4.19.0-6-cloud-amd64 _#1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11) x86_64 GNU/Linux
-----> memory: 
                     total        used        free      shared  buff/cache   available
       Mem:            989         139         475           2         374         713
       Swap:          4095           0        4095
-----> docker version: 
       Client: Docker Engine - Community
        Version:           19.03.5
        API version:       1.40
        Go version:        go1.12.12
        Git commit:        633a0ea838
        Built:             Wed Nov 13 07:25:38 2019
        OS/Arch:           linux/amd64
        Experimental:      false
       
       Server: Docker Engine - Community
        Engine:
         Version:          19.03.5
         API version:      1.40 (minimum version 1.12)
         Go version:       go1.12.12
         Git commit:       633a0ea838
         Built:            Wed Nov 13 07:24:09 2019
         OS/Arch:          linux/amd64
         Experimental:     false
        containerd:
         Version:          1.2.10
         GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
        runc:
         Version:          1.0.0-rc8+dev
         GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
        docker-init:
         Version:          0.18.0
         GitCommit:        fec3683
-----> docker daemon info: 
WARNING: No swap limit support
       Client:
        Debug Mode: true
       
       Server:
        Containers: 0
         Running: 0
         Paused: 0
         Stopped: 0
        Images: 46
        Server Version: 19.03.5
        Storage Driver: overlay2
         Backing Filesystem: extfs
         Supports d_type: true
         Native Overlay Diff: true
        Logging Driver: json-file
        Cgroup Driver: cgroupfs
        Plugins:
         Volume: local
         Network: bridge host ipvlan macvlan null overlay
         Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
        Swarm: inactive
        Runtimes: runc
        Default Runtime: runc
        Init Binary: docker-init
        containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
        runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
        init version: fec3683
        Security Options:
         apparmor
         seccomp
          Profile: default
        Kernel Version: 4.19.0-6-cloud-amd64
        Operating System: Debian GNU/Linux 10 (buster)
        OSType: linux
        Architecture: x86_64
        CPUs: 1
        Total Memory: 989.9MiB
        Name: statica
        ID: IWAL:6WPQ:26JJ:OXQZ:Y2YG:FI5B:VM7O:G6S7:XOIV:Q5IF:JZOG:SJT7
        Docker Root Dir: /var/lib/docker
        Debug Mode: false
        Registry: https://index.docker.io/v1/
        Labels:
        Experimental: false
        Insecure Registries:
         127.0.0.0/8
        Live Restore Enabled: false
       
-----> sigil version: 0.4.0
-----> herokuish version: 
       herokuish: 0.5.3
       buildpacks:
         heroku-buildpack-multi     v1.0.0
         heroku-buildpack-ruby      v201
         heroku-buildpack-nodejs    v161
         heroku-buildpack-clojure   v84
         heroku-buildpack-python    v154
         heroku-buildpack-java      v63
         heroku-buildpack-gradle    v30
         heroku-buildpack-scala     v85
         heroku-buildpack-play      v26
         heroku-buildpack-php       v159
         heroku-buildpack-go        v126
         buildpack-nginx            v12
-----> dokku version: dokku version 0.19.10
-----> dokku plugins: 
       plugn: 0.3.2
         00_dokku-standard    0.19.10 enabled    dokku core standard plugin
         20_events            0.19.10 enabled    dokku core events logging plugin
         app-json             0.19.10 enabled    dokku core app-json plugin
         apps                 0.19.10 enabled    dokku core apps plugin
         builder-dockerfile   0.19.10 enabled    dokku core builder-dockerfile plugin
         builder-herokuish    0.19.10 enabled    dokku core builder-herokuish plugin
         buildpacks           0.19.10 enabled    dokku core buildpacks plugin
         certs                0.19.10 enabled    dokku core certificate management plugin
         checks               0.19.10 enabled    dokku core checks plugin
         common               0.19.10 enabled    dokku core common plugin
         config               0.19.10 enabled    dokku core config plugin
         docker-options       0.19.10 enabled    dokku core docker-options plugin
         domains              0.19.10 enabled    dokku core domains plugin
         elasticsearch        1.9.2 enabled    dokku elasticsearch service plugin
         enter                0.19.10 enabled    dokku core enter plugin
         git                  0.19.10 enabled    dokku core git plugin
         http-auth            0.4.0 enabled    HTTP authentication for apps
         letsencrypt          0.9.1 enabled    Automated installation of let's encrypt TLS certificates
         logs                 0.19.10 enabled    dokku core logs plugin
         maintenance          0.5.0 enabled    Maintenance mode for apps
         network              0.19.10 enabled    dokku core network plugin
         nginx-vhosts         0.19.10 enabled    dokku core nginx-vhosts plugin
         plugin               0.19.10 enabled    dokku core plugin plugin
         postgres             1.9.2 enabled    dokku postgres service plugin
         proxy                0.19.10 enabled    dokku core proxy plugin
         ps                   0.19.10 enabled    dokku core ps plugin
         redirect             0.6.2 enabled    Plugin for managing application redirects
         redis                1.10.1 enabled    dokku redis service plugin
         repo                 0.19.10 enabled    dokku core repo plugin
         resource             0.19.10 enabled    dokku core resource plugin
         scheduler-docker-local 0.19.10 enabled    dokku core scheduler-docker-local plugin
         shell                0.19.10 enabled    dokku core shell plugin
         ssh-keys             0.19.10 enabled    dokku core ssh-keys plugin
         storage              0.19.10 enabled    dokku core storage plugin
         tags                 0.19.10 enabled    dokku core tags plugin
         tar                  0.19.10 enabled    dokku core tar plugin
         trace                0.19.10 enabled    dokku core trace plugin
```

No, sadly with the [latest Docker 19.03.6](https://github.com/docker/docker-ce/releases) it's still an issue:

```
# dokku elasticsearch:create lollipop
=====> Extracting config files
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x5613698273f8]

goroutine 1 [running]:
github.com/docker/cli/cli/command.ValidateOutputPath(0x7ffdc961f11c, 0x1, 0xc0003100c0, 0x20)
	/go/src/github.com/docker/cli/cli/command/utils.go:142 +0x118
```

--- a/functions
+++ b/functions
@@ -80,14 +80,14 @@ service_create_container() {
   else
     dokku_log_info2 "Extracting config files"
     local TEMP_DOCKER_CONTAINER_ID=$(docker create "$PLUGIN_IMAGE:$PLUGIN_IMAGE_VERSION")
-    docker cp "${TEMP_DOCKER_CONTAINER_ID}:/usr/share/elasticsearch/config/" - | tar -x -C "$SERVICE_HOST_ROOT/config" --strip-components=1
+    docker cp "${TEMP_DOCKER_CONTAINER_ID}:/usr/share/elasticsearch/config/" "$SERVICE_HOST_ROOT/"
     if [[ "$ELASTICSEARCH_MAJOR_VERSION" -eq "5" ]]; then
       docker cp "${TEMP_DOCKER_CONTAINER_ID}:/etc/elasticsearch/jvm.options" - | tar -x -C "$SERVICE_HOST_ROOT/config"
     fi
     docker rm -v "${TEMP_DOCKER_CONTAINER_ID}"
-    sed -i.bak 's#Xms2g#Xms512m#g; s#Xmx2g#Xmx512m#g' "$SERVICE_HOST_ROOT/config/jvm.options"
+    sed -i.bak 's#Xms1g#Xms512m#g; s#Xmx1g#Xmx512m#g' "$SERVICE_HOST_ROOT/config/jvm.options"
 
-    ID=$(docker run --name "$SERVICE_NAME" -v "$SERVICE_HOST_ROOT/data:/usr/share/elasticsearch/data" -v "$SERVICE_HOST_ROOT/config:/usr/share/elasticsearch/config" -v "$SERVICE_HOST_ROOT/config:/usr/share/elasticsearch/config" --env-file="$SERVICE_ROOT/ENV" -d --restart always --label dokku=service --label dokku.service=elasticsearch "$PLUGIN_IMAGE:$PLUGIN_IMAGE_VERSION")
+    ID=$(docker run --name "$SERVICE_NAME" -v "$SERVICE_HOST_ROOT/data:/usr/share/elasticsearch/data" -v "$SERVICE_HOST_ROOT/config:/usr/share/elasticsearch/config" --env-file="$SERVICE_ROOT/ENV" -d --restart always --label dokku=service --label dokku.service=elasticsearch "$PLUGIN_IMAGE:$PLUGIN_IMAGE_VERSION")
     echo "$ID" >"$SERVICE_ROOT/ID"
 
     if [[ "$ELASTICSEARCH_MAJOR_VERSION" -eq "5" ]]; then

